MapsAndPlacesDemo/MapsAndPlacesDemo/OverlayController.swift

```diff
@@ -0,0 +1,145 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import UIKit
+import GooglePlaces
+import GoogleMaps
+
+class OverlayController {
+    
+    private var overlays = [GMSCircle]()
+    private var lat: Double = 0.0
+    private var long: Double = 0.0
+    
+    // MARK: Methods to get the placeID from a set of coordinates
+    
+    /// Searches an API for the entire data JSON file based on the lat and long values
+    ///
+    /// - Parameter completion: The completion handler.
+    func fetchData(completion: @escaping ([String : Any]?, Error?) -> Void) {
+        let apiKey: String = ApiKeys.mapsAPI
+        let url =  "https://maps.googleapis.com/maps/api/geocode/json?&latlng=\(lat),\(long)&key="
+        let search = URL(string: url + apiKey)!
+        let task = URLSession.shared.dataTask(with: search) { (data, response, error) in
+            guard let data = data else { return }
+            do {
+                if let array = try JSONSerialization.jsonObject(
+                    with: data,
+                    options: .allowFragments
+                    ) as? [String : Any] {
+                    completion(array, nil)
+                }
+            } catch {
+                print(error)
+                completion(nil, error)
+            }
+        }
+        task.resume()
+    }
+    
+    /// Finds and parses data to extract a placeId from a set of coordinates and sends the appropriate data to the completion handler
+    ///
+    /// - Parameters:
+    ///   - latitude: The latitude of the location.
+    ///   - longitude: The longitude of the location.
+    ///   - completion: The completion handler.
+    func geocode(
+        latitude: Double,
+        longitude: Double,
+        completion: @escaping (_ placemark: [CLPlacemark]?, _ error: Error?, _ pid: String) -> Void
+    ) {
+        CLGeocoder().reverseGeocodeLocation(
+            CLLocation(
+                latitude: latitude,
+                longitude: longitude
+            )
+        ) { placemark, error in
+            guard let placemark = placemark, error == nil else {
+                completion(nil, error, "")
+                return
+            }
+            self.lat = latitude
+            self.long = longitude
+            var ans: String = ""
+            self.fetchData { (dict, error) in
+                let convert = String(describing: dict?["results"])
+                var counter: Int = 0
+                var characters = [Character]()
+                let search = "place_id"
+                for letter in search {
+                    characters.append(letter)
+                }
+                var startPlaceId: Bool = false
+                for ch in convert {
+                    if !startPlaceId {
+                        if ch == characters[counter] {
```

Comment:
Here, I guaranteed the array isn't empty since the code manually appends each letter of "place_id" into the characters array before going into this loop.